pypit/bpmimage.py

```diff
@@ -59,48 +62,39 @@ class BPMImage(object):
     bpm : ndarray
 
     """
-    # Keep order same as processimages (or else!)
-    def __init__(self, spectrograph=None, settings=None, det=None,
-                 binning=None, reduce_badpix=None, msbias=None, shape=None,
-                 spectro_class=None):
 
-        # Parameters unique to this Object
-        self.spectrograph = spectrograph
-        self.binning = binning
-        self.det = det
-        self.spectro_class = spectro_class
+    # Frametype is a class attribute
+    frametype = 'bpm'
 
-        self.reduce_badpix = reduce_badpix
-        self.msbias = msbias
+    # Keep order same as processimages (or else!)
+    def __init__(self, spectrograph=None, shape=None, filename=None, det=None, msbias=None,
+                 trim=True):
+
+        # Spectrograph is required
+        if isinstance(spectrograph, basestring):
```

Comment:
I thought we decided that spectrograph would only be an instance of the spectrograph class?

Reply:
This was my middle ground solution that I didn't change post Tuesday's discussion.  This kind of `if..elif..else` block for the spectrograph is everywhere.  I'm again going to punt on this.  It's easy enough to pull it out in a later PR.